src/Microsoft.AspNetCore.Server.HttpSys/AsyncAcceptContext.cs

```diff
@@ -184,6 +184,7 @@ internal uint QueueBeginGetContext()
 
         internal void AllocateNativeRequest(uint? size = null, ulong requestId = 0)
         {
+            _nativeRequestContext?.ReleasePins();
```

Comment:
I'm surprised that nativeRequestContext's Dispose does not Release it's own pins...

Reply:
I guess we can't change that?

Reply:
It can, but it follows a usage pattern where it it wants to make sure things happen in a specific order. This is more of a "you're not using me the way I want you to" issue than a functional problem.

Reply:
We can, but it's a bigger refactor than I have bandwidth for at the moment.

Reply:
Got it, thanks.